src/test/util/blockfilter.cpp

```diff
@@ -12,7 +12,7 @@
 bool ComputeFilter(BlockFilterType filter_type, const CBlockIndex* block_index, BlockFilter& filter)
 {
     CBlock block;
-    if (!ReadBlockFromDisk(block, block_index->GetBlockPos(), Params().GetConsensus())) {
+    if (!ReadBlockFromDisk(block, WITH_LOCK(cs_main, return block_index->GetBlockPos()), Params().GetConsensus())) {
```

Comment:
nit: unit tests are single-threaded, so you can just take the lock for the whole function. It is likely that there are other lock violations anyway: See https://github.com/bitcoin/bitcoin/issues/17161#issuecomment-543151879

Reply:
done